### PR TITLE
fix: remove unnecessary filter_map usages

### PR DESCRIPTION
**Pull request by wowinter13**
fixes https://github.com/rust-lang/rust-clippy/issues/12556

(Fixed version of https://github.com/rust-lang/rust-clippy/pull/12766)

changelog: [unnecessary_filter_map]: filter map improvements

### CODE REVIEW

File: clippy_lints/src/methods/unnecessary_filter_map.rs [hunk 1]
```diff
@@ -52,15 +70,18 @@ pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx hir::Expr<'tcx>, a
...
             cx,
             if name == "filter_map" {
                 UNNECESSARY_FILTER_MAP
             } else {
                 UNNECESSARY_FIND_MAP
             },
             expr.span,
-            format!("this `.{name}` can be written more simply using `.{sugg}`"),
+            format!("this `.{name}` can be written more simply"),
+            format!("try replacing with `.{sugg}`"),
```
Comment (llogiq) [inline clippy_lints/src/methods/unnecessary_filter_map.rs:82]: This doubles the suggestion in the output. Please reduce the text to "try instead" or "replace with".

**Review by llogiq — COMMENTED** [review]
The suggestion can be shortened, otherwise this looks good.

File: clippy_lints/src/methods/unnecessary_filter_map.rs (hunk 1 above)
Comment (wowinter13) [reply clippy_lints/src/methods/unnecessary_filter_map.rs]: @llogiq fixed, thanks!

**Comment by llogiq** [comment]
Thank you! @bors r+

**Pull request merged**